catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/CswTransformProvider.java

```diff
@@ -146,11 +148,15 @@ private void writeXml(BinaryContent content, HierarchicalStreamWriter writer) {
      */
     @Override
     public Object unmarshal(HierarchicalStreamReader reader, UnmarshallingContext context) {
-        Object outputSchema = context.get(CswConstants.OUTPUT_SCHEMA_PARAMETER);
-        Object typeName = context.get(CswConstants.TYPE_NAME_PARAMETER);
+        String outputSchema = (String) context.get(CswConstants.OUTPUT_SCHEMA_PARAMETER);
+        String typeName = (String) context.get(CswConstants.TYPE_NAME_PARAMETER);
+        String keyArg = (String) context.get(CswConstants.TRANSFORMER_LOOKUP_KEY);
```

Comment:
These are never used. They should replace outputSchema & typeName to allow the caller to specify which one and the value the same way it works in the marshal method.